src/components/homepage/Homepage.tsx

```diff
@@ -1,20 +1,45 @@
 import React from 'react';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import { RootState } from '../../state/store';
+import { selectResource } from '../../state/actions/resourceActions';
+import { Link } from 'react-router-dom';
+import { DataFetched } from '../../state/reducers/resource';
+import { requestResource } from '../../state/thunkMiddleware';
 
 const Homepage: React.FC<{}> = () => {
-    const stateApp  = useSelector((state: RootState) => state);
-    if (stateApp.resource.loading === true){
+    const stateApp = useSelector((state: RootState) => state);
```

Comment:
Try to get only the data you need. Because if you don't, when the reducer state will be big you will be importing a lot of useless data into the component.